crates/matrix-sdk-indexeddb/src/crypto_store/migrations.rs

```diff
@@ -0,0 +1,138 @@
+// Copyright 2023 The Matrix.org Foundation C.I.C.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use indexed_db_futures::{prelude::*, web_sys::DomException};
+use tracing::info;
+use wasm_bindgen::JsValue;
+
+use crate::crypto_store::{keys, Result};
+
+/// Open the indexeddb with the given name, upgrading it to the latest version
+/// of the schema if necessary.
+pub async fn open_and_upgrade_db(name: &str) -> Result<IdbDatabase, DomException> {
```

Comment:
(looking at the commit ` Move migration functions to a separate module and split it up into smaller functions`)

In the future, can you avoid changes that both 1. move the code, 2. change the code, please? They make the review really hard. In this case the amount of code is small enough that it's not a big deal, but it takes quadratically more time than if the commit had been split into two smaller logical ones.

Reply:
Noted, and agreed. Sorry about this!